lib/helpers.js

```diff
@@ -61,6 +63,58 @@ function overrideTags (parent, child, telegraf) {
   return toAppend.length > 0 ? tags.concat(toAppend) : tags;
 }
 
+/**
+ * Overrides tags in parent with tags from child with the same name (case sensitive) and return the result as a string.
+ */
+function overrideTagsToStringUnoptimized(parent, child, telegraf, separator) {
+  const tags = overrideTagsUnoptimized(parent, child, telegraf);
+  return tags.join(separator);
+}
+
+/**
+ * Overrides tags in parent with tags from child with the same name (case sensitive) and return the result as a string.
+ *
+ * More performant than overrideTagsUnoptimized. But it requires that globalTags and child are both objects, not arrays.
+ */
+function overrideTags2 (globalTags, globalTagsFullMemo, child, telegraf, separator) {
+  let result = '';
+  const usedGlobalKeys = new Set();
+  // eslint-disable-next-line require-jsdoc
+  function addToResult(tag, addToFront) {
+    if (result === '') {
+      result += tag;
+    }
+    // this is just here to match the behavior of overrideTagsUnoptimized
+    else if (addToFront) {
+        result = tag + separator + result;
+      }
+      else {
+        result += separator + tag;
+      }
+  }
+  for (const c of Object.keys(child)) {
+    // there is a global tag with the same name as the child tag - use child
+    const formattedChildKey = sanitizeTags(c);
+    if (Object.hasOwn(globalTags, formattedChildKey)) {
+      usedGlobalKeys.add(formattedChildKey);
+    }
+    const serializedTagWithValue = `${formattedChildKey}:${sanitizeTags(child[c], telegraf)}`;
+    addToResult(serializedTagWithValue);
+  }
+  if (usedGlobalKeys.size === 0) {
+    addToResult(globalTagsFullMemo, true);
+  }
+  else {
+    for (const g of Object.keys(globalTags)) {
+      if (!usedGlobalKeys.has(g)) {
+        const serializedTagWithValue = `${g}:${sanitizeTags(globalTags[g], telegraf)}`;
+        result = serializedTagWithValue + separator + result;
```

Comment:
should we use `addToResult(serializedTagWithValue, true)` here?

Reply:
ya